crates/sui-core/src/authority/transaction_deferral.rs

```diff
@@ -90,11 +90,15 @@ pub fn transaction_deferral_within_limit(
         deferred_from_round,
     } = deferral_key
     {
-        return (future_round - deferred_from_round) <= max_deferral_rounds_for_congestion_control;
+        if let Some(diff) = future_round.checked_sub(*deferred_from_round) {
```

Comment:
I think this is not necessary because `deferred_from_round` can never be greater than `future_round`

If you wanted to add an invariant check for extra safety,  rather than doing it here, I would put a `debug_fatal!` log in `DeferralKey::new_for_consensus_round` if the values are wrong, and then for release builds make it bump the value of `future_round` to be at least `deferred_from_round`

Reply:
I agree this cannot happen, but i'm not opposed to being more defensive here. I'd suggest a `debug_assert(future_round > deferred_from_round)` and then a saturating sub

Reply:
I've made the updates to the PR as you suggested. please take a look!